src/main/java/com/tabnine/lifecycle/BinaryStateService.java

```diff
@@ -0,0 +1,47 @@
+package com.tabnine.lifecycle;
+
+import com.intellij.openapi.Disposable;
+import com.intellij.openapi.application.ApplicationManager;
+import com.intellij.util.messages.MessageBus;
+import com.intellij.util.messages.MessageBusConnection;
+import com.tabnine.binary.BinaryRequestFacade;
+import com.tabnine.binary.requests.config.StateRequest;
+import com.tabnine.binary.requests.config.StateResponse;
+import com.tabnine.general.DependencyContainer;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+public class BinaryStateService implements Disposable {
+
+    private static final ScheduledExecutorService scheduler = Executors.newScheduledThreadPool(1);
+    private final BinaryRequestFacade binaryRequestFacade = DependencyContainer.instanceOfBinaryRequestFacade();
+    private final MessageBus messageBus;
+    private StateResponse lastStateResponse;
+
+    public BinaryStateService() {
+        this.messageBus = ApplicationManager.getApplication().getMessageBus();;
+        scheduler.scheduleAtFixedRate(this::updateState, 0, 1, TimeUnit.SECONDS);
+    }
+
+    public StateResponse getLastStateResponse() {
```

Comment:
Not sure I understand what you mean. I save the lastStateResponse so that I can compare it to the new one and notify listeners only if it changed.
This getter is for immediate retrieval of the last known value.
Makes sense?